### DISCUSSION TITLE
Signing in with third party

### DISCUSSION THREAD

**Discussion by MatD1** [rphlmr/supa-fly-stack#71] [Q&A] [answered]
Hi everyone, a huge thanks for this awesome template it's been super helpful. I've been primarily using this template to learn how to mix Supabase with Remix, I wanted to inquire if it was possible to sign user's in with a third party such as `Github`. I did attempt to write this myself however the `mapAuthSession` function that is used in the other sign in methods requires a session object that `signInWithOAuth` does not seem to export.

```
export async function signInWithGithub() {
  const { data, error } = await getSupabaseAdmin().auth.signInWithOAuth({
    provider: "github",
  });

// session does not exist

  if (!data.session || error) return null;

  return mapAuthSession(data.session);
}
```

Any tips would be greatly appreciated!

**Comment by rphlmr** [comment, marked as answer]
Hello, Thanks for your kind words :)

To make it working, you have to follow the[ Supabase doc](https://supabase.com/docs/guides/auth/social-login/auth-github) but I guess you did.

Then, in you Supabase Dashboard, allow redirect urls `/oauth/callback` (one per domain)

![CleanShot 2023-03-20 at 09 35 21@2x](https://user-images.githubusercontent.com/20722140/226287016-f4661fa5-a4ba-4b1d-84f4-cde4e4d1ac21.png)

And finally, take some inspiration from this branch :p https://github.com/rphlmr/supa-fly-stack/compare/main...example/github-auth

You have to trigger 3rd party auth from the browser and tell Supabase Auth server where to redirect after success.

Whether magic link or social provider, it all ends up in `/oauth/callback`.

**Reply by MatD1** [reply]
Huge thanks for that, I had a feeling I should have used client side but was unsure how it would work with the callback. Appreciate the quick response!
